tools/wpt/update_chromium_revision.py

```
@@ -0,0 +1,88 @@
+    with open("tools/wpt/latest_chromium_revision.txt", "r") as f:
+        existing_revision = int(f.read())
+
+    # Find the lowest new revision number among latest revisions by platform.
+    # We need to find a revision number that is available for download for all platforms,
+    # so we start looking from the smallest of these latest revisions.
+    smallest_revision = existing_revision
+    for platform, filename in PLATFORM_INFO:
+        try:
+            url = f"{SNAPSHOTS_PATH}{platform}/LAST_CHANGE"
+            revision = get(url).text.strip()
+            smallest_revision = max(smallest_revision, int(revision))
+        except requests.RequestException as e:
+            print(f"failed LAST_CHANGE lookup for {platform}: {e}")
+            continue
+
+    if smallest_revision == existing_revision:
+        return
+
+    # Step backwards through revision numbers until we find one
+    # that is available for all platforms.
+    largest_mutually_available_revision = smallest_revision
+    available_for_all = False
+    timed_out = False
+    while not available_for_all and not timed_out:
+        available_for_all = True
+        # For each platform, check if Chromium is available for download from snapshots.
+        for platform, filename in PLATFORM_INFO:
+            try:
+                url = (f"{SNAPSHOTS_PATH}{platform}/"
+                       f"{largest_mutually_available_revision}/{filename}")
+                # Check the headers of each possible download URL.
+                r = requests.head(url)
+                # If the "Accept-Ranges" header is not present, we know the file is not
+                # available for download. Decrement the revision number and try again.
+                if "Accept-Ranges" not in r.headers:
+                    largest_mutually_available_revision -= 1
+                    available_for_all = False
+                    break
+            except requests.RequestException as e:
+                print(e)
```

Can you add a "bad thing happened when..." style description, that makes it clear whether it's safe to ignore this error? Or should we just let the script fail when there's a network error?

---

Changed to better reflect what is happening 👍